### DISCUSSION TITLE
Command conflicting with ZI (zsh plugin manager)

### DISCUSSION THREAD

**Discussion by Kritiqual** [ajeetdsouza/zoxide#446] [Q&A] [answered]
Is there a way to disable `zi` command, it conflict with this [z-shell/zi](https://github.com/z-shell/zi) which also use zi command

**Comment by ajeetdsouza** [comment, marked as answer]
You can initialize it with no commands and then alias `__zoxide_z`:

```zsh
eval "$(zoxide init zsh --no-cmd)"
alias z=__zoxide_z
```

**Reply by Kritiqual** [reply]
I forgot to put `"` in `eval`

**Reply by ngocphamm** [reply]
@ajeetdsouza One thing I just figured out with `--no-cmd` option is that I would lose the ability to have fzf autocompletion with `z phrase<SPACE><TAB>`, because the `__zoxide_z_complete` function will not be generated in that case.

Would there be a possibility that I can also customize both the commands `z` and `zi` and not just the prefix? I'm currently still using `z` as the main command, and `zz` for the `zi` version.

**Reply by ajeetdsouza** [reply]
Fixed in https://github.com/ajeetdsouza/zoxide/commit/86741bbe6a34b30ae632dee09ff17a7a68b62a2e. In the next release, if you define an alias on `__zoxide_z`, it will include completions.

**Comment by ngocphamm** [comment]
@ajeetdsouza It seems that version 0.9.4 does not put the `__zoxide_z_complete()` again with `--no-cmd` option specified. My `z phrase<SPACE><TAB>` stops working after updating to that version.

**Reply by ngocphamm** [reply]
@ajeetdsouza This is still a problem with version 0.9.5 just released.

I have tried to have my `.zshrc` contains only these lines and `z Foo<Space><TAB>` still doesn't work. `z Foo` still works, taking me into that folder.

```
autoload -U compinit; compinit

eval "$(zoxide init zsh --no-cmd)"

alias z=__zoxide_z
alias zz=__zoxide_zi
```

**Reply by ajeetdsouza** [reply]
Added back in https://github.com/ajeetdsouza/zoxide/commit/3d3267b4fd73e6292d317f96d60ef69ea035912d

**Reply by ngocphamm** [reply]
@ajeetdsouza Thanks for the fix, but for some reason, the new version `0.9.6` still doesn't work for me.

1. I change my `.zshrc` to contain only the above
2. Start a new shell, `rm .zcompdump`
3. Start a new shell again
    * Run `z Qu` -> works, taking me to the folder
    * Run `zz<enter>` -> works, showing me fzf selection
    * Run `z Q<space><tab>` -> doesn't work, not showing anything

Version `0.9.3` works fine for all the 3 situations above. Specifically case 3 I get fzf selection filtered with `Q`.

**Reply by ajeetdsouza** [reply]
Aliases in zsh don't support completions by default. Either use a function like so:

```sh
function z() {
    __zoxide_z "$@"
}
```

Or, enable completions on aliases:

```sh
setopt completealiases
```

**Reply by ngocphamm** [reply]
@ajeetdsouza Thank you for the guidance! After playing with those for a bit, I found out it works for me with both `setopt completealiases` and having `compdef __zoxide_z_complete z` *after* 

```
alias z=__zoxide_z
alias zz=__zoxide_zi
```

**Reply by ngocphamm** [reply]
@ajeetdsouza I looked at this a little more today because the `setopt completealiases` line seems to mess a few other things for me.

Comparing the output of `zoxide init zsh --no-cmd` between version `0.9.3` and `0.9.6`, I see this line being removed. And if I add it back, things work fine, without having to set either `setopt completealiases` and `compdef __zoxide_z_complete z` like I mentioned above.

`[[ "${+functions[compdef]}" -ne 0 ]] && \compdef __zoxide_z_complete __zoxide_z`
